### PR TITLE
[Email Routing] Add Email Worker guidelines in _index.md

### PR DESCRIPTION
**Pull request by jahands**
Using Email Workers can cause unexpected bounced emails in some cases that developers may not be aware of. To help ensure email delivery isn't impacted, we're adding some guidelines/best practices for writing Email Workers.

### CODE REVIEW

File: content/email-routing/email-workers/_index.md [hunk 1]
```diff
@@ -41,6 +41,14 @@ The route, or email address, bound to the Worker forwards emails to your Email W
 
 For example, say that you create an allowlist Email Worker and bind it to a `hello@my-company.com` route. This route will be the email address you share with the world, to make sure that only email addresses on your allowlist are forwarded to your destination address. All other emails will be dropped.
 
+## Guidelines
+
+It's important to keep the following in mind when writing Email Workers:
+
+1. Unhandled exceptions will result in a permanent SMTP failures. Email Workers should generally have a top-level `try/catch` block to prevent throwing unhandled exceptions.
```
Comment (edevil) [inline content/email-routing/email-workers/_index.md]: I'm actually changing this in the next release. Having a top-level try/catch, perhaps even with Sentry, is a good practice. But worker exceptions will result in a temporary smtp error back to the client.

File: content/email-routing/email-workers/_index.md [hunk 2]
```diff
@@ -45,9 +45,8 @@ For example, say that you create an allowlist Email Worker and bind it to a `hel
...
-3. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending email to addresses which bounce too often. So it's important to not cause unexpected bounces when writing Email Workers.
+1. Unhandled exceptions will result in temporary SMTP failures, which typically results in the sender retrying later. If you don't want senders retrying, consider adding a top-level `try/catch` block to ensure there are no unhandled exceptions. Some senders may stop sending email to addresses with too many temporary failures.
+2. Email Workers that take longer than 50 seconds to respond will result in a permanent SMTP failure. If some work can be done in the background, consider using [context.waitUntil()](/workers/runtime-apis/handlers/fetch/#contextwaituntil) to execute background tasks after the Email Worker returns. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending email to addresses which bounce too often. So it's important to not cause unexpected bounces when writing Email Workers.
```
Comment (edevil) [inline content/email-routing/email-workers/_index.md:49]: These will be handled as exceptions as well, hence temporary smtp errors.

File: content/email-routing/email-workers/_index.md [hunk 3]
```diff
@@ -41,6 +41,13 @@ The route, or email address, bound to the Worker forwards emails to your Email W
 
 For example, say that you create an allowlist Email Worker and bind it to a `hello@my-company.com` route. This route will be the email address you share with the world, to make sure that only email addresses on your allowlist are forwarded to your destination address. All other emails will be dropped.
 
+## Guidelines
+
+It's important to keep the following in mind when writing Email Workers:
```
Comment (deadlypants1973) [inline content/email-routing/email-workers/_index.md:46]:
```suggestion
Note the following when writing Email Workers:
```

File: content/email-routing/email-workers/_index.md [hunk 4]
```diff
@@ -41,6 +41,13 @@ The route, or email address, bound to the Worker forwards emails to your Email W
...
+
+1. Unhandled exceptions will result in temporary SMTP failures, which typically results in the sender retrying later. If you don't want senders retrying, consider adding a top-level `try/catch` block to ensure there are no unhandled exceptions. Some senders may stop sending email to addresses with too many temporary failures.
```
Comment (deadlypants1973) [inline content/email-routing/email-workers/_index.md:48]:
```suggestion
1. Unhandled exceptions will result in temporary SMTP failures, which typically result in the sender retrying later. If you do not want senders retrying, add a top-level `try/catch` block to ensure there are no unhandled exceptions. Some senders may stop sending emails to addresses with too many temporary failures.
```

File: content/email-routing/email-workers/_index.md [hunk 5]
```diff
@@ -41,6 +41,13 @@ The route, or email address, bound to the Worker forwards emails to your Email W
...
+2. Email Workers that take longer than 50 seconds to respond will result in a permanent SMTP failure. If some work can be done in the background, consider using [context.waitUntil()](/workers/runtime-apis/handlers/fetch/#contextwaituntil) to execute background tasks after the Email Worker returns. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending email to addresses which bounce too often. So it's important to not cause unexpected bounces when writing Email Workers.
```
Comment (deadlypants1973) [inline content/email-routing/email-workers/_index.md:49]:
```suggestion
2. Email Workers that take longer than 50 seconds to respond will result in a permanent SMTP failure. If some work can be done in the background, consider using [`context.waitUntil()`](/workers/runtime-apis/handlers/fetch/#contextwaituntil) to execute background tasks after the Email Worker returns. Senders will typically consider permanent SMTP errors to be a bounce, and may stop sending emails to addresses which bounce too often. Therefore, it is important to not cause unexpected bounces when writing Email Workers.
```

**Review by deadlypants1973 — COMMENTED** [review]
PCX

**Comment by kodster28** [comment]
Closing this out for now, but moving to an internal ticket for tracking.
PCX-10691 for reference

**Pull request closed without merging**